Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -448,16 +477,31 @@ public final class RegistryClient {
 
         // We either use a previously recorded checksum, or fetch it from the registry
         func withExpectedChecksum(body: @escaping (Result<String, Error>) -> Void) {
-            if let expectedChecksum = expectedChecksum {
-                return body(.success(expectedChecksum))
+            if let fingerprintStorage = self.fingerprintStorage {
+                fingerprintStorage.get(package: package, version: version, kind: .registry,
```

Comment:
nit: in this code base we normally format each argument in its own new line when calling methods, i.e.

```
fingerprintStorage.get(
  package: package, 
  version: version, 
  kind: .registry,
  observabilityScope: observabilityScope, 
  callbackQueue: callbackQueue
) { result in
...
}
```